File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/inline/comment/direct/Java/Go/s185737640_commented.go
package main

import (
	"fmt"
)

func main() {
	var s string
	fmt.Scanln(&s)

	if len(s)%2 == 1 {
		s = s[:len(s)-1]
	} else {
		s = s[:len(s)-2]
	}

	for len(s) > 0 {
		if isEvenString(s) {
			break
		}
		s = s[:len(s)-2]
	}
	fmt.Println(len(s))
}

// 長さが偶数の文字列sが遇文字列かどうかを返す
func isEvenString(s string) bool {
	a := s[:len(s)/2]
	b := s[len(s)/2:]
	return a == b
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/comment/uncomment/direct/Java/Python/s185737640_commented.py
def is_even_string(s):
    a = s[:len(s)//2]
    b = s[len(s)//2:]
    return a == b

def main():
    import sys
    input = sys.stdin.read().strip()
    
    if len(input) % 2 == 1:
        input = input[:-1]
    else:
        input = input[:-2]
    
    while len(input) > 0:
        if is_even_string(input):
            break
        input = input[:-2]
    
    print(len(input))

if __name__ == "__main__":
    main()
# End of Code
